src/transformers/models/encodec/modeling_encodec.py

```diff
@@ -762,6 +803,24 @@ def decode(
             return (audio_values,)
         return EncodecDecoderOutput(audio_values)
 
+    def compute_mel_spectrogram(self, audio, n_fft, hop_length, n_mels=64):
+        device = audio.device
+        # Adjust n_mels if necessary to avoid warnings
+        n_mels = min(n_mels, n_fft // 2 + 1)
+        # Create the window function on the correct device
+        window = torch.hann_window(n_fft, device=device)
+        mel_spec_transform = torchaudio.transforms.MelSpectrogram(
```

Comment:
If we're using `torchaudio`, we need to add this as a soft dependency by using `is_torchaudio_available`, like shown [here](https://github.com/huggingface/transformers/blob/4e90b99ed916300b80bac9db793f2a96b2a87122/src/transformers/models/musicgen_melody/feature_extraction_musicgen_melody.py#L33-L34)